packages/material-ui-lab/src/useAutocomplete/useAutocomplete.js

```diff
@@ -585,7 +587,7 @@ export default function useAutocomplete(props) {
               inputRef.current.value.length,
             );
           }
-        } else if (freeSolo && inputValueFilter !== '') {
```

Comment:
Introduced in #18786. 

> The Enter key shouldn't trigger a new onChange event if its already currently selected.

-- https://github.com/mui-org/material-ui/issues/18344#issuecomment-553360657

For future reference: Prefer to codify this explicitly rather than communicating this implicitly with a type (here: the empty string). A variable `if (inputValueIsSelectedValue === false)` is more readable than `if (inputValueFilter === '')`

Reply:
Sounds fair